Set up Docker, dockerized Django app

Desferatum opened the pull request:
- set up docker files (Dockerfile, docker-compose.yml)
- dockerized existing Django app (forum core) - server runs successfully 
- changed sqlite3 to postgresql in settings.py

mehalyna commented on docker-compose.yml:
```diff
@@ -0,0 +1,34 @@
+      sh -c "cd forum &&
+             python manage.py migrate &&
+             python manage.py runserver 0.0.0.0:8000"
+    environment:
+      - DEBUG=1
+      - DB_NAME=postgres
+      - DB_USER=postgres
+      - DB_PASSWORD=postgres
+      - DB_HOST=db
+      - DB_PORT=5432
```
Use an `.env` file instead of hardcoding credentials
```
env_file:
  - .env
```
`.env`:
```
POSTGRES_DB=postgres
POSTGRES_USER=postgres
POSTGRES_PASSWORD=postgres
DB_NAME=postgres
DB_USER=postgres
DB_PASSWORD=postgres
DB_HOST=db
DB_PORT=5432
DEBUG=1
```
Then reference the same variables in Django `settings.py`.

mehalyna commented on docker-compose.yml:
```diff
@@ -0,0 +1,34 @@
+      - "8000:8000"
+    depends_on:
+      - db
+    command: >
+      sh -c "cd forum &&
+             python manage.py migrate &&
+             python manage.py runserver 0.0.0.0:8000"
```
Avoid running migrations directly in the command. It's better to separate concerns using `entrypoint.sh`.
```
command: sh /app/entrypoint.sh
```
And in `entrypoint.sh`:
```
#!/bin/sh

echo "Applying migrations..."
python manage.py migrate

echo "Collecting static files..."
python manage.py collectstatic --noinput

echo "Starting server..."
python manage.py runserver 0.0.0.0:8000
```

mehalyna commented on docker-compose.yml:
```diff
@@ -0,0 +1,34 @@
+      - DB_HOST=db
+      - DB_PORT=5432
+
+  db:
+    image: postgres:15
+    environment:
+      - POSTGRES_DB=postgres
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+    volumes:
+      - postgres_data:/var/lib/postgresql/data/
```
Define `healthcheck` for `db` service, to ensure DB is really ready before `app` starts:
```
healthcheck:
  test: ["CMD-SHELL", "pg_isready -U $$POSTGRES_USER -d $$POSTGRES_DB"]
  interval: 10s
  retries: 5
```

Desferatum commented:
- updated .env.example with credentials to show how local .env file should look like (.env is in .gitignore)
- updated docker-compose.yml: 
- - now using variables from .env (must be in root to work properly, see .env.example); 
- - added healthcheck to db service 
- added entrypoint.sh with commands to run at the start:
- - migrate
- - collectstatic
- - runserver
- configured STATICROOT in settings.py for correct work of collectstatic
- added staticfiles/ to .gitignore
